[24기 이송현]Add: Footer 컴포넌트 리팩토링

vi2920va opened the pull request:
>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
(여기에 수정 사항에 대한 간략한 한줄 요약/제목 작성해 주세요.)
- icon index.html에서 fontAwesome는 stylesheet를 연결하여 사용.
-  공통으로 사용되는 common.scss에서 `margin`, `padding` 불필요한 초기화 코드 제거
- body 태그에  전체적인 폰트 스타일을 `font-family: Arial, Helvetica, sans-serif` 추가.
- 전체스크린 화면일 경우, 데스크탑 일 경우 두 가지 사이즈 고려하여 레이아웃 페이지 구현.

- 스크린 화면 
![pullpage](https://i.postimg.cc/hj9m4PVn/Screenshot-from-2021-08-31-15-18-43.png)

- 데스크탑 화면
![pullpage](https://i.postimg.cc/QMzmKtBW/Screenshot-from-2021-08-31-15-18-57.png)
		 
## 수정 사항들 자세한 내용
(여기에 수정 사항에 대한 자세한 내용을 작성해 주세요.)
- 스타일 적용하지 않는 불필요한 선택자 제거.
- height 속성을 준 요소 수정.
- 미디어 쿼리에서 불필요한 스타일 코드 제거.

## 기타 질문 및 특이 사항
(궁금한 사항들, 하면서 느낀 점들 공유해주실 것이 있으면 자유롭게 작성해 주세요.)


## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

vi2920va committed:
Add: 공통 푸터 컴포넌트 레이아웃 추가
 - 공통으로 사용되는 푸터 컴포넌트 레이아웃 완성

vi2920va committed:
Modify: margin, padding 초기화 코드 제거

vi2920va committed:
Add: fontAwesome 적용

vi2920va committed:
Modify: import 경로 수정

vi2920va committed:
Modify: margin, padding 초기화 제거

saengmotmi commented:
@vi2920va 
- `[황대웅] Add: 장바구니 페이지 완료` 와 같은 형태로 PR 제목 변경해주세요
- PR 메시지에 적어주신 내용만 보고서는 작업하신 내용을 파악하기 어려운 것 같습니다. 리뷰어가 해당 PR에 대한 맥락을 알 수 있도록 작업하신 내용을 설명해주시고, 필요하다면 스크린샷 같은 시각 자료를 추가해주시면 더욱 좋습니다. cdn을 통해 fontAwesome 같은 라이브러리도 추가하셨고 공용 스타일도 수정하셨으니 더더욱 해당 변경 사항에 대한 설명이 필요합니다. 모두 문서화 역량과 관련된 연습이라고 생각하시고 꼼꼼하게 적어주세요. 이후 리뷰 다시 보겠습니다.

vi2920va commented:
> @vi2920va
> 
> * `[황대웅] Add: 장바구니 페이지 완료` 와 같은 형태로 PR 제목 변경해주세요
> * PR 메시지에 적어주신 내용만 보고서는 작업하신 내용을 파악하기 어려운 것 같습니다. 리뷰어가 해당 PR에 대한 맥락을 알 수 있도록 작업하신 내용을 설명해주시고, 필요하다면 스크린샷 같은 시각 자료를 추가해주시면 더욱 좋습니다. cdn을 통해 fontAwesome 같은 라이브러리도 추가하셨고 공용 스타일도 수정하셨으니 더더욱 해당 변경 사항에 대한 설명이 필요합니다. 모두 문서화 역량과 관련된 연습이라고 생각하시고 꼼꼼하게 적어주세요. 이후 리뷰 다시 보겠습니다.



> @vi2920va
> 
> * `[황대웅] Add: 장바구니 페이지 완료` 와 같은 형태로 PR 제목 변경해주세요
> * PR 메시지에 적어주신 내용만 보고서는 작업하신 내용을 파악하기 어려운 것 같습니다. 리뷰어가 해당 PR에 대한 맥락을 알 수 있도록 작업하신 내용을 설명해주시고, 필요하다면 스크린샷 같은 시각 자료를 추가해주시면 더욱 좋습니다. cdn을 통해 fontAwesome 같은 라이브러리도 추가하셨고 공용 스타일도 수정하셨으니 더더욱 해당 변경 사항에 대한 설명이 필요합니다. 모두 문서화 역량과 관련된 연습이라고 생각하시고 꼼꼼하게 적어주세요. 이후 리뷰 다시 보겠습니다.

ㄴ[

vi2920va commented:
대웅님 리뷰 참고하여 다시 수정했습니다.

saengmotmi commented on src/components/Footer/Footer.scss:
```diff
@@ -0,0 +1,403 @@
+    background: #fff;
+  }
+
+  .aboutWrapper {
+    h4 {
+      margin-bottom: 14px;
+      font-size: 9px;
+      font-weight: 600;
+      color: #545454;
+    }
+  }
+
+  .collectionWrapper {
+    h4 {
+      margin-bottom: 14px;
+      font-size: 9px;
+      font-weight: 600;
+      color: #545454;
+    }
+
+  }
+
+  .helpWrapper {
+    h4 {
+      margin-bottom: 14px;
+      font-size: 9px;
+      font-weight: 600;
+      color: #545454;
+    }
+  }
```
```suggestion
  h4 { // 혹은 .menuTitle 과 같이 클래스 셀렉터 사용
    margin-bottom: 14px;
    font-size: 9px;
    font-weight: 600;
    color: #545454;
  }
```
동일한 스타일에 대해서 코드가 중복되고 있는 것 같습니다. 특별한 이유가 없다면 동일한 스타일은 동일한 코드로 한번에 표현하는 것이 어떨까요?

saengmotmi commented on src/components/Footer/Footer.scss:
```diff
@@ -0,0 +1,403 @@
+
+
+
+@media (max-width: 937px) {
```
미디어 쿼리는 한 뎁스 안쪽으로 넣어주시고 기존 속성에 비해 변경되는 요소만 적어주셔도 될 것 같습니다.

saengmotmi commented on src/components/Footer/Footer.scss:
```diff
@@ -0,0 +1,403 @@
+// @media (max-width:1200px) {
+.footerContainer {
+  .iconWrapper {
+    .iconList {
+      display: flex;
+      justify-content: space-around;
```
```suggestion
.footerContainer {
    .iconList {
      display: flex;
      justify-content: space-around;
```
스타일이 없는 셀렉터까지 적어줄 필요 없습니다

saengmotmi commented on src/components/Footer/Footer.scss:
```diff
@@ -0,0 +1,403 @@
+        border-left: 0;
+      }
+
+      li {
+        height: 15px;
```
거의 대부분의 경우 고정 높이 주실 이유가 없습니다. 자식 요소의 크기 + padding, margin 방식으로 레이아웃 잡아주세요

saengmotmi commented on src/components/Footer/Footer.scss:
```diff
@@ -0,0 +1,403 @@
+          height: 20px;
+        }
+
+        .langeBtn {
+          span {
+            position: relative;
```
전반적으로 불필요한 셀렉터 중첩이 보입니다. 이 경우에도 `.langeContent`로 직접 사용해줄 수 있겠네요.
```suggestion
        .langeContent {
            position: relative;
```

vi2920va replied:
종택님 리뷰 반영하여 수정했습니다

saengmotmi reviewed (commented):
PR 설명 잘 써주셨네요! 코멘트 확인해주시고 다음에는 리뷰 요청 라벨 달아두시는 것도 잊지 말아주세요.

vi2920va committed:
Modify: 푸터 컴포넌트 스타일 수정
- Footer.scss에서 불필요한 중첩 선택자 제거
- 미디어 쿼리에서 불필요한 스타일 코드 제거
- li 태그 height 적용 부분 수정

vi2920va committed:
Modify: 미디어 쿼리 수정

saengmotmi commented:
@vi2920va 고생하셨습니다 🔥
